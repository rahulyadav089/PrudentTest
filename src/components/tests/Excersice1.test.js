import { fireEvent, render } from "@testing-library/react";
import Excersice1 from "../Excersice1";
import testdictionary from "./testdictionary";

describe("excersice 1 component", () => {
  test("should autocorrect and give correct result", () => {
    const { container, getByTestId } = render(
      <Excersice1 dictionary={testdictionary} />
    );
    const input = getByTestId("maininput");
    expect(input).toBeInTheDocument();
    fireEvent.change(input, { target: { value: "cannot" } });
    expect(input.value).toBe("cannot");

    const submitBtn = getByTestId("submit");
    expect(submitBtn).toBeInTheDocument();

    fireEvent.click(submitBtn);

    const resultInput = getByTestId("resultinput");

    expect(resultInput.value).toBe("cannot");
  });

  test("should autocorrect and  fail", () => {
    const { container, getByTestId } = render(
      <Excersice1 dictionary={testdictionary} />
    );
    const input = getByTestId("maininput");
    expect(input).toBeInTheDocument();
    fireEvent.change(input, { target: { value: "maybea" } });
    expect(input.value).toBe("maybea");

    const submitBtn = getByTestId("submit");
    expect(submitBtn).toBeInTheDocument();

    fireEvent.click(submitBtn);

    expect(getByTestId("fail-alert")).toBeInTheDocument();
  });

  test("should reset the input", () => {
    const { getByTestId } = render(<Excersice1 dictionary={testdictionary} />);
    const input = getByTestId("maininput");
    expect(input).toBeInTheDocument();
    fireEvent.change(input, { target: { value: "maybea" } });
    expect(input.value).toBe("maybea");

    const resetBtn = getByTestId("reset");

    fireEvent.click(resetBtn);

    expect(input.value).toBe("");
  });
});
