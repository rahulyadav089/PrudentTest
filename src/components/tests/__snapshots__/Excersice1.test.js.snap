// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`excersice 1 component should autocorrect 1`] = `
<div>
  <div>
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <h5
          class="card-title"
        >
          Auto Correct Checker
        </h5>
      </div>
      <div
        class="card-body"
      >
        <div
          class="input-group"
        >
          <input
            class="form-control"
            data-testid="maininput"
            placeholder="Enter a word"
            type="text"
            value=""
          />
          <button
            class="btn btn-success"
            data-testid="check"
            type="button"
          >
            <div>
              check if word exist
            </div>
          </button>
        </div>
      </div>
      <div
        class="card-body"
      >
        <button
          class="btn btn-warning"
          data-testid="reset"
          type="button"
        >
          Reset
        </button>
      </div>
    </div>
  </div>
</div>
`;
